src/wsproxy.ml

```diff
@@ -45,7 +46,15 @@ let start path handler =
          Lwt_unix.close fd_sock2 >>= fun _ ->
```

Comment:
We don't protect closing `fd_sock2` with an `Lwt_finalize` block. If an exception happens in any of the unprotected lines, I think it will be left open, and the program will not terminate, we will just continue the loop due to the surrounding `Lwt.catch` block. So I think this way we can leak open sockets if we fail many times before we close `fd_sock2`.
However, I can see that we also close `fd_sock2` in a try/with block above, so if none of the unprotected lines can raise an exception, than the current code is fine (but hard to reason about).